tests/bsim/bluetooth/host/gatt/ccc_value/testlib/att_read.c

```diff
@@ -0,0 +1,126 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/__assert.h>
+
+#include "att_read.h"
+
+struct bt_testlib_att_read_closure {
+	uint8_t att_err;
+	struct bt_conn *conn;
+	struct bt_gatt_read_params params;
+	uint16_t *result_size;
+	uint16_t *result_handle;
+	struct net_buf_simple *result_data;
+	struct k_mutex lock;
+	struct k_condvar done;
+};
+
+static inline uint8_t att_read_cb(struct bt_conn *conn, uint8_t att_err,
+				  struct bt_gatt_read_params *params, const void *read_data,
+				  uint16_t read_len)
+{
+	struct bt_testlib_att_read_closure *ctx =
+		CONTAINER_OF(params, struct bt_testlib_att_read_closure, params);
+
+	k_mutex_lock(&ctx->lock, K_FOREVER);
+
+	ctx->att_err = att_err;
+
+	if (!att_err && ctx->result_handle) {
+		*ctx->result_handle = params->by_uuid.start_handle;
+	}
+
+	if (!att_err && ctx->result_size) {
+		*ctx->result_size = read_len;
+	}
+
+	if (!att_err && ctx->result_data) {
+		uint16_t result_data_size =
+			MIN(read_len, net_buf_simple_tailroom(ctx->result_data));
+		net_buf_simple_add_mem(ctx->result_data, read_data, result_data_size);
+	}
```

Comment:
Consider having a `if (!att_err)` with the other 3 `if` insides of it. Not really any reason to check `att_err` 3 times